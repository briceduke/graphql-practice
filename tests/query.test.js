const gql = require("graphql-tag");
const createTestServer = require("./helper");
const FEED = gql`
  {
    feed {
      id
      message
      createdAt
      likes
      views
    }
  }
`;

const POSTS = gql`
  {
    posts {
      message
    }
  }
`;

describe("queries", () => {
  test("feed", async () => {
    const { query } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          findMany: jest.fn(() => [
            {
              id: 1,
              message: "hello",
              createdAt: 12345839,
              likes: 20,
              views: 300,
            },
          ]),
        },
      },
    });

    const res = await query({ query: FEED });
    expect(res).toMatchSnapshot();
  });

  test("posts", async () => {
    const { query } = createTestServer({
      user: { id: 1 },
      models: {
        User: {
          posts: jest.fn(() => [
            {
              id: 1,
              message: "hello",
              createdAt: 12345839,
              likes: 20,
              views: 300,
            },
          ]),
        },
      },
    });

    const res = await query({ query: POSTS });
    expect(res).toMatchSnapshot();
  });
});
