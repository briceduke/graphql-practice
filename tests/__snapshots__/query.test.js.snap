// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "createdAt": "12345839",
        "id": "1",
        "likes": 20,
        "message": "hello",
        "views": 300,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries posts 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Cannot read property 'findMany' of undefined],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
